src/Passes/Source/Apps/Qat/Config.hpp

```diff
@@ -0,0 +1,82 @@
+#pragma once
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+#include "Commandline/ConfigurationManager.hpp"
+
+namespace microsoft
+{
+namespace quantum
+{
+
+    /// Main configuration class for the qat command-line program.
+    class QatConfig
+    {
+      public:
+        using String = std::string;
+
+        // Functions required by configuration manager
+        //
+
+        /// Setup function that binds instance variables to the command-line/configuration entries.
+        /// This function also provide descriptions of each of the properties below.
+        void setup(ConfigurationManager& config);
+
+        // Flags and options
+        //
+
+        /// List of dynamic libraries to load.
+        String load() const;
+
+        /// Flag that indicates whether or not we are generating a new QIR by applying a profile.
+        bool generate() const;
+
+        /// Flag to indicate whether or not to verify that the (Q)IR is a valid LLVM IR.
+        bool verifyModule() const;
+
+        /// Flag to indicate whether or not to validate the compliance with the QIR profile.
+        bool validate() const;
+
+        /// String to request a specific profile name. Default is base.
+        String profile() const;
+
+        /// Indicates whether or not the QIR adaptor tool should emit LLVM IR to the standard output.
+        bool emitLlvm() const;
+
+        /// Enables optimisation level 0. Note higher OX override lower ones.
+        bool opt0() const;
+
+        /// Enables optimisation level 1. Note higher OX override lower ones.
+        bool opt1() const;
+
+        /// Enables optimisation level 2. Note higher OX override lower ones.
+        bool opt2() const;
+
+        /// Enables optimisation level 3. Note higher OX override lower ones.
+        bool opt3() const;
+
```

Comment:
1.
> Note higher OX override lower ones

This comment tells me that the optimisation level values are mutually exclusive (i.e. `opt0()` and `opt2()` cannot both return `true`). If that's the case then consider merging these four into a single call that returns the optimisation level (either as an integer or as an `enum` or in some other way, but only a single active optimisation level value).

2.
> /// Enables optimisation level 0.

Based on the implementation, intuitively I feel that you meant to write "/// Tells if the optimisation level 0 is enabled".
In that case consider renaming `opt0()` to something like `isOpt0Enabled()`.
Applicable to a number of member functions in this class.

Reply:
Updated accordingly.